test/bntests.pl

```diff
@@ -0,0 +1,157 @@
+#! /usr/bin/env perl
+# Copyright 2008-2016 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+# Run the tests specified in bntests.txt, as a check against OpenSSL.
+use strict;
+use warnings;
+use Math::BigInt;
+
+sub bn()
+{
+    my $x = shift;
+    my ($sign, $hex) = ($x =~ /^([+\-]?)(.*)$/);
+
+    $hex = '0x' . $hex if $hex !~ /^0x/;
+    return Math::BigInt->from_hex($sign.$hex);
+}
+
+sub evaluate()
+{
+    my ( $lineno ) = shift;
+    my ( %s ) = @_;
+
+    if ( defined $s{'Sum'} ) {
+        # Sum = A + B
+        my $sum = &bn($s{'Sum'});
+        my $a = &bn($s{'A'});
+        my $b = &bn($s{'B'});
+        return if $sum == $a + $b;
+    } elsif ( defined $s{'LShift1'} ) {
+        # LShift1 = A * 2
+        my $lshift1 = &bn($s{'LShift1'});
+        my $a = &bn($s{'A'});
+        return if $lshift1 == $a->bmul(2);
+    } elsif ( defined $s{'LShift'} ) {
+        # LShift = A * 2**N
+        my $lshift = &bn($s{'LShift'});
+        my $a = &bn($s{'A'});
+        my $n = &bn($s{'N'});
+        return if $lshift == $a->blsft($n);
+    } elsif ( defined $s{'RShift'} ) {
+        # RShift = A / 2**N
+        my $rshift = &bn($s{'RShift'});
+        my $a = &bn($s{'A'});
+        my $n = &bn($s{'N'});
+        return if $rshift == $a->brsft($n);
+    } elsif ( defined $s{'Square'} ) {
+        # Square = A * A
+        my $square = &bn($s{'Square'});
+        my $a = &bn($s{'A'});
+        return if $square == $a->bmul($a);
+    } elsif ( defined $s{'Product'} ) {
+        # Product = A * B
+        my $product = &bn($s{'Product'});
+        my $a = &bn($s{'A'});
+        my $b = &bn($s{'B'});
+        return if $product == $a->bmul($b);
+    } elsif ( defined $s{'Quotient'} ) {
+        # Quotient = A / B
+        # Remainder = A - B * Quotient
+        my $quotient = &bn($s{'Quotient'});
+        my $remainder = &bn($s{'Remainder'});
+        my $a = &bn($s{'A'});
+        my $b = &bn($s{'B'});
+
+        # First the remainder test.
+        $b->bmul($quotient);
+        my $rempassed = $remainder == $a->bsub($b) ? 1 : 0;
+
+        # XXX TODO BROKEN
```

Comment:
Which do you mean is broken in this regard, what Math::BigInt or OpenSSL BN?  Mathematically, Math::BigInt is more correct, but OpenSSL BN behaves the same way as bc, and may be what people expect after all these years...  So the question is what you'd like to see changed.  I'm thinking you should mention that explicitely.